File: backend/app/schemas/document.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional



class DocumentResponse(BaseModel):
    id: int
    filename: str
    file_path: str
    upload_status: str
    owner_id: int

    pages: Optional[int] = None

    title: Optional[str] = None

    author: Optional[str] = None

    file_size: Optional[int] = None

    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/document.ts
export interface DocumentItem {
  id: number;
  filename: string;
  file_path: string;
  upload_status: string;
  owner_id: number;
  pages: number | null;
  title: string | null;
  author: string | null;
  file_size: number;
  created_at: string;
}